docs/snippets/gettingStarted/webApi/apiBDD.e2e.spec.ts

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import {
+  getInMemoryEventStore,
+  type EventStore,
+} from '@event-driven-io/emmett';
+import {
+  ApiE2ESpecification,
+  expectResponse,
+  getApplication,
+} from '@event-driven-io/emmett-expressjs';
+import { randomUUID } from 'node:crypto';
+import { beforeEach, describe, it } from 'node:test';
+import type { PricedProductItem } from '../events';
+import { ShoppingCartStatus } from './shoppingCart';
+import { shoppingCartApi } from './simpleApi';
+
+const getUnitPrice = () => {
+  return Promise.resolve(100);
+};
+
+describe('ShoppingCart E2E', () => {
+  let clientId: string;
+  let shoppingCartId: string;
+
+  beforeEach(() => {
+    clientId = randomUUID();
+    shoppingCartId = `shopping_cart:${clientId}:current`;
+  });
+
+  describe('When empty', () => {
+    it('should add product item', () => {
+      return given((request) =>
+        request
+          .post(`/clients/${clientId}/shopping-carts/current/product-items`)
+          .send(productItem),
+      )
+        .when((request) =>
+          request.get(`/clients/${clientId}/shopping-carts/current`).send(),
```

Comment:
I've created this temporal workaround `get` method - I thought that it would be good to do this and in the future we just need to change the underlying implementation detail to use projections

Reply:
Yes, sounds good to me, I like the idea. Thanks to that, we'll have the full coverage.